configure.ac

```diff
@@ -89,6 +89,8 @@ fi
 # Check whether to build a with debug symbols
 LIBZMQ_CHECK_ENABLE_DEBUG
 
+AC_CHECK_LIB([atomic],[__atomic_load_4],[LIBS="${LIBS} -latomic"]) 
```

Comment:
This will always enable this linking, which is not right. There's a section that checks the host os, and looks for android, you should add it there instead with the appropriate checks for "android things"